boot/bootstate20.go

```diff
@@ -587,3 +690,65 @@ func (bsmark *bootState20MarkSuccessful) commit() error {
 
 	return nil
 }
+
+// genericEarlyBootChooseSnap will run the logic to choose which snap should be
+// mounted during the early boot, i.e. initramfs. Specifically, it only works
+// with kernel and base snaps. It returns the first and second choice for what
+// snaps to mount, if second is set, then it is the primary fallback snap, and
+// the first snap is the try snap, if second is unset, then first is the primary
+// fallback snap. It returns both so that a higher level function can do
+// additional verification of the try snap if it wants, such as the kernel snaps
+// being verified in the modeenv, or the base snaps updating the modeenv with
+// new values for base_status, etc.
```

Comment:
I notice that the comment doesn't mention the expectedTryStatus, I also wonder if that should be more something like expectFallback bool and let the caller deal with what it means

Reply:
done, returning expectFallback does make the usages slightly simpler